dev/run-tests.py

```diff
@@ -595,13 +601,28 @@ def main():
 
     changed_modules = None
     changed_files = None
-    if test_env == "amplab_jenkins" and os.environ.get("AMP_JENKINS_PRB"):
+    should_only_test_modules = "TEST_ONLY_MODULES" in os.environ
+    included_tags = []
+    if should_only_test_modules:
+        str_test_modules = [m.strip() for m in os.environ.get("TEST_ONLY_MODULES").split(",")]
+        test_modules = [m for m in modules.all_modules if m.name in str_test_modules]
+        # Directly uses test_modules as changed modules to apply tags and environments
+        # as if all specified test modules are changed.
+        changed_modules = test_modules
+        str_excluded_tags = os.environ.get("TEST_ONLY_EXCLUDED_TAGS", None)
+        str_included_tags = os.environ.get("TEST_ONLY_INCLUDED_TAGS", None)
+        excluded_tags = []
+        if str_excluded_tags:
+            excluded_tags = [t.strip() for t in str_excluded_tags.split(",")]
+        included_tags = []
+        if str_included_tags:
+            included_tags = [t.strip() for t in str_included_tags.split(",")]
+    elif test_env == "amplab_jenkins" and os.environ.get("AMP_JENKINS_PRB"):
         target_branch = os.environ["ghprbTargetBranch"]
         changed_files = identify_changed_files_from_git_commits("HEAD", target_branch=target_branch)
         changed_modules = determine_modules_for_files(changed_files)
         excluded_tags = determine_tags_to_exclude(changed_modules)
-
-    if not changed_modules:
+    else:
```

Comment:
shall we keep the original logic `if not changed_modules:` here?

Reply:
I think the logic is same as before.

Reply:
The original logical is to run all the test if `changed_modules` is empty. And current logic is to run all the tests if it is not from github action or AMP lab jenkins.
But this is trivial as everything is working now.

Reply:
Oh, okay. let me just change it back